### DISCUSSION TITLE
Help with clear_color.rs example

### DISCUSSION THREAD

**Discussion by crumplecup** [bevyengine/bevy#2635] [Q&A] [answered]
Hi all,

I am new to Rust and Bevy.  The error has me flummoxed, any help is appreciated.

The example headless.rs runs for me, but anything that opens a window gets the following error:

bevy/examples on 🌱 latest via 🦀 v1.54.0 took 12s
✗  RUST_BACKTRACE=1 cargo +nightly run --example clear_color
    Finished dev [unoptimized + debuginfo] target(s) in 0.14s
     Running `/home/erik/crates/bevy/target/debug/examples/clear_color`
WARNING: lavapipe is not a conformant vulkan implementation, testing use only.
thread 'main' panicked at 'Failed to acquire next swap chain texture!', crates/bevy_wgpu/src/renderer/wgpu_render_resource_context.rs:381:18
stack backtrace:
   0: rust_begin_unwind
             at /rustc/b70888601af92f6cdc0364abab3446e418b91d36/library/std/src/panicking.rs:515:5
   1: core::panicking::panic_fmt
             at /rustc/b70888601af92f6cdc0364abab3446e418b91d36/library/core/src/panicking.rs:92:14
   2: core::option::expect_failed
             at /rustc/b70888601af92f6cdc0364abab3446e418b91d36/library/core/src/option.rs:1578:5
   3: core::option::Option<T>::expect
             at /rustc/b70888601af92f6cdc0364abab3446e418b91d36/library/core/src/option.rs:685:21
   4: <bevy_wgpu::renderer::wgpu_render_resource_context::WgpuRenderResourceContext as bevy_render::renderer::render_resource_context::RenderResourceContext>::next_swap_chain_texture
             at /home/erik/crates/bevy/crates/bevy_wgpu/src/renderer/wgpu_render_resource_context.rs:380:13
   5: <bevy_render::render_graph::nodes::window_swapchain_node::WindowSwapChainNode as bevy_render::render_graph::node::Node>::update
             at /home/erik/crates/bevy/crates/bevy_render/src/render_graph/nodes/window_swapchain_node.rs:68:34
   6: bevy_wgpu::renderer::wgpu_render_graph_executor::WgpuRenderGraphExecutor::execute
             at /home/erik/crates/bevy/crates/bevy_wgpu/src/renderer/wgpu_render_graph_executor.rs:75:25
   7: bevy_wgpu::wgpu_renderer::WgpuRenderer::run_graph::{{closure}}
             at /home/erik/crates/bevy/crates/bevy_wgpu/src/wgpu_renderer.rs:115:13
   8: bevy_ecs::world::World::resource_scope
             at /home/erik/crates/bevy/crates/bevy_ecs/src/world/mod.rs:695:22
   9: bevy_wgpu::wgpu_renderer::WgpuRenderer::run_graph
             at /home/erik/crates/bevy/crates/bevy_wgpu/src/wgpu_renderer.rs:104:9
  10: bevy_wgpu::wgpu_renderer::WgpuRenderer::update
             at /home/erik/crates/bevy/crates/bevy_wgpu/src/wgpu_renderer.rs:121:9
  11: bevy_wgpu::get_wgpu_render_system::{{closure}}
             at /home/erik/crates/bevy/crates/bevy_wgpu/src/lib.rs:112:9
  12: <alloc::boxed::Box<F,A> as core::ops::function::FnMut<Args>>::call_mut
             at /rustc/b70888601af92f6cdc0364abab3446e418b91d36/library/alloc/src/boxed.rs:1579:9
  13: <bevy_ecs::system::exclusive_system::ExclusiveSystemFn as bevy_ecs::system::exclusive_system::ExclusiveSystem>::run
             at /home/erik/crates/bevy/crates/bevy_ecs/src/system/exclusive_system.rs:41:9
  14: <bevy_ecs::schedule::stage::SystemStage as bevy_ecs::schedule::stage::Stage>::run
             at /home/erik/crates/bevy/crates/bevy_ecs/src/schedule/stage.rs:812:25
  15: bevy_ecs::schedule::Schedule::run_once
             at /home/erik/crates/bevy/crates/bevy_ecs/src/schedule/mod.rs:201:13
  16: <bevy_ecs::schedule::Schedule as bevy_ecs::schedule::stage::Stage>::run
             at /home/erik/crates/bevy/crates/bevy_ecs/src/schedule/mod.rs:219:21
  17: bevy_app::app::App::update
             at /home/erik/crates/bevy/crates/bevy_app/src/app.rs:58:9
  18: bevy_winit::winit_runner_with::{{closure}}
             at /home/erik/crates/bevy/crates/bevy_winit/src/lib.rs:485:17
  19: winit::platform_impl::platform::sticky_exit_callback
             at /home/erik/.cargo/registry/src/github.com-1ecc6299db9ec823/winit-0.24.0/src/platform_impl/linux/mod.rs:736:5
  20: winit::platform_impl::platform::x11::EventLoop<T>::run_return
             at /home/erik/.cargo/registry/src/github.com-1ecc6299db9ec823/winit-0.24.0/src/platform_impl/linux/x11/mod.rs:303:17
  21: winit::platform_impl::platform::x11::EventLoop<T>::run
             at /home/erik/.cargo/registry/src/github.com-1ecc6299db9ec823/winit-0.24.0/src/platform_impl/linux/x11/mod.rs:398:9
  22: winit::platform_impl::platform::EventLoop<T>::run
             at /home/erik/.cargo/registry/src/github.com-1ecc6299db9ec823/winit-0.24.0/src/platform_impl/linux/mod.rs:652:56
  23: winit::event_loop::EventLoop<T>::run
             at /home/erik/.cargo/registry/src/github.com-1ecc6299db9ec823/winit-0.24.0/src/event_loop.rs:154:9
  24: bevy_winit::run
             at /home/erik/crates/bevy/crates/bevy_winit/src/lib.rs:171:5
  25: bevy_winit::winit_runner_with
             at /home/erik/crates/bevy/crates/bevy_winit/src/lib.rs:493:9
  26: bevy_winit::winit_runner
             at /home/erik/crates/bevy/crates/bevy_winit/src/lib.rs:211:5
  27: core::ops::function::Fn::call
             at /rustc/b70888601af92f6cdc0364abab3446e418b91d36/library/core/src/ops/function.rs:70:5
  28: <alloc::boxed::Box<F,A> as core::ops::function::Fn<Args>>::call
             at /rustc/b70888601af92f6cdc0364abab3446e418b91d36/library/alloc/src/boxed.rs:1586:9
  29: bevy_app::app::App::run
             at /home/erik/crates/bevy/crates/bevy_app/src/app.rs:68:9
  30: bevy_app::app_builder::AppBuilder::run
             at /home/erik/crates/bevy/crates/bevy_app/src/app_builder.rs:49:9
  31: clear_color::main
             at ./window/clear_color.rs:4:5
  32: core::ops::function::FnOnce::call_once
             at /rustc/b70888601af92f6cdc0364abab3446e418b91d36/library/core/src/ops/function.rs:227:5
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
Aug 10 20:31:12.058 ERROR gpu_alloc::block: Memory block wasn't deallocated
Aug 10 20:31:12.058 ERROR gpu_alloc::block: Memory block wasn't deallocated
fish: “RUST_BACKTRACE=1 cargo +nightly…” terminated by signal SIGSEGV (Address boundary error)

**Comment by bjorn3** [comment]
What GPU do you use? Do you have a vulkan driver installed for your GPU?

**Reply by crumplecup** [reply]
As recently as late June, I was able to run some simple 3D examples on this particular machine, but when I came back to it, something has changed. Very likely a self-inflicted wound of some kind.

Vulkan Instance Version: 1.2.131


VGA compatible controller: Advanced Micro Devices, Inc. [AMD/ATI] Juniper XT [Radeon HD 5770] (prog-if 00 [VGA controller])
        Subsystem: PC Partner Limited / Sapphire Technology Juniper XT [Radeon HD 5770]
        Physical Slot: 2
        Flags: bus master, fast devsel, latency 0, IRQ 28
        Memory at e0000000 (64-bit, prefetchable) [size=256M]
        Memory at f7e20000 (64-bit, non-prefetchable) [size=128K]
        I/O ports at e000 [size=256]
        Expansion ROM at 000c0000 [disabled] [size=128K]
        Capabilities: <access denied>
        Kernel driver in use: radeon
        Kernel modules: radeon

**Comment by The-DevBlog** [comment]
I was getting a very similar error. Can you post a snippet of your 'use' statements?

**Reply by crumplecup** [reply]
On my end, it looks like:

```rust
git clone https://github.com/bevyengine/bevy
cd bevy
cargo run --example clear_color
```
clear_color.rs
```rust
use bevy::{prelude::*, render::pass::ClearColor};

fn main() {
    App::new()
        .insert_resource(ClearColor(Color::rgb(0.5, 0.5, 0.9)))
        .add_plugins(DefaultPlugins)
        .run();
}
```

**Reply by The-DevBlog** [reply]
Weird. I was having the same issue related to 'bevy_wgpu' when I was using the 'bevy::prelude::window' (i think thats it). I ended up removing it for now to temporarily solve the issue. I would assume its related to the ClearColor. Have you removed that just to see if it will compile?

**Reply by crumplecup** [reply]
If I change `DefaultPlugins` to `MinimalPlugins`, then it will compile without error and without further effect.

**Reply by The-DevBlog** [reply]
Strange. I honestly don't know. I'll look around some more and hopefully find an answer. This same thing was driving me nuts a few days ago.

**Comment by crumplecup** [comment, marked as answer]
This was an upstream issue and Bevy was not at fault.  I ended up switching distros to Arch and I am able to run all examples without issue.
